### PR TITLE
Update to current versions.

### PR DESCRIPTION
**Pull request by icemac**
Especially Sphinx got a bunch of new dependencies where requests seems to be
the heaviest one as it has a handful of dependencies itself.

### CODE REVIEW

**Comment by hannosch** [comment]
Looks like zope.browserpage, menu and resource all depend on test internals from `zope.component.testfiles.views` and those have changed. Since it's three different projects depending on them, maybe it's easier to change `zope.component`.

**Comment by jamadden** [comment]
> Since it's three different projects depending on them, maybe it's easier to change zope.component.

I personally would be -1 on that. I'm not a fan of having external projects depend on details in a package that the package itself doesn't even touch (that's why `testfiles` was changed---nothing in `zope.component` used it) and which aren't documented. There was nothing there that was really specific to `zope.component` or that depended on `zope.component` implementation details---it was just a set of (trivial) generic classes. Better to make the other projects self-contained, IMO.

**Comment by jamadden** [comment]
> Better to make the other projects self-contained

And I'll volunteer for that. I was going to be touching those projects in the next few days (hopefully!) for PyPy3 and Python 3.6 support anyway.

**Comment by hannosch** [comment]
> And I'll volunteer for that.

+1 and cheers :) I was just being pragmatic.

**Comment by jamadden** [comment]
There are now PRs for all three of those projects.

- zopefoundation/zope.browserpage#3
- zopefoundation/zope.browsermenu#6
- zopefoundation/zope.browserresource#4

**Comment by icemac** [comment]
The broken buildout runs are fixed in zopefoundation/zope.browserresource#5.

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ python:
     - 3.6
     - pypy
 install:
-    - pip install -U setuptools==33.1.1
+    - pip install -U setuptools==`grep setuptools ztk-versions.cfg | awk '{print $3}'`
```
Comment (mgedmin) [inline .travis.yml]: You could do this with just awk: `$(awk '$1 == "setuptools" {print $3}' ztk-versions.cfg)`

Comment (icemac) [reply .travis.yml]: Cool idea, I never use awk.

**Pull request merged**